src/ExpireNFTv4.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+
+contract ExpireNFTv4 is ERC721, ERC721Enumerable, ERC721Burnable, AccessControl {
+    using Counters for Counters.Counter;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+
+    uint256 public constant EXPIRE_TIME_IN_SECONDS = 60 * 60 * 24 * 365;
+    uint256 public constant SUBSCRIPTION_PRICE = 0.10 ether;
+
+    mapping(uint256 => uint256) public expireTime;
+
+    constructor() ERC721("ExpireNFTv4", "EXPD4") {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _grantRole(MINTER_ROLE, msg.sender);
+    }
+
+    function safeMint(address to) public onlyRole(MINTER_ROLE) {
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
+
+        require(!_exists(tokenId), "ExpireNFTv3: invalid token ID");
+
+        _safeMint(to, tokenId);
+
+        expireTime[tokenId] = block.timestamp + EXPIRE_TIME_IN_SECONDS;  
+    }
+
+    function addOneYearOfSubscription(uint256 tokenId) public payable {
+        require(!_exists(tokenId), "ExpireNFTv4: invalid token ID");
```

Comment:
Look at this `require`, I think the condition is the opposite, the tokenId is supposed to exist, right?

Reply:
Fixed 😊